Sources/Widgets/Representations/LineHandleRepresentation/index.js

```diff
@@ -64,6 +67,99 @@ function vtkLineHandleRepresentation(publicAPI, model) {
     model.displayMapper.setCallback(callback ? callbackProxy : null);
   };
 
+  const superPublicAPI = { ...publicAPI };
+  publicAPI.requestData = (inData, outData) => {
+    superPublicAPI.requestData(inData, outData);
+    if (!model.holeWidth) {
+      return;
+    }
+    const internalPolyData = outData[0];
+
+    // Duplicate points and point data
+    const points = internalPolyData.getPoints();
+    const dataArrays = internalPolyData.getPointData().getArrays();
+    [points, ...dataArrays].forEach((array) => {
+      const oldNumberOfValues = array.getNumberOfValues();
+      array.resize(2 * array.getNumberOfTuples());
+      const arrayData = array.getData();
+      for (let i = 0; i < oldNumberOfValues; ++i) {
+        arrayData[i + oldNumberOfValues] = arrayData[i];
+      }
+    });
+
+    // Change the scale and origin of each line
+    const states = publicAPI.getRepresentationStates(inData[0]);
+    const nStates = states.length;
+    const nGlyphs = 2 * nStates;
+    let scaleArray = internalPolyData.getPointData().getArrayByName('scale');
+    if (scaleArray?.getNumberOfTuples() !== nGlyphs) {
+      model._pipeline.mapper.setScaleArray('scale');
+      model._pipeline.mapper.setScaleFactor(1);
+      model._pipeline.mapper.setScaling(true);
+      model._pipeline.mapper.setScaleMode(
+        vtkGlyph3DMapper.ScaleModes.SCALE_BY_COMPONENTS
+      );
+      scaleArray = allocateArray(
+        internalPolyData,
+        'scale',
+        nGlyphs,
+        'Float32Array',
+        3
+      );
+      const scale = [10, 10, 1000];
+      const scaleData = scaleArray.getData();
+      for (let i = 0; i < scaleData.length; i += scale.length) {
+        scaleData.set(scale, i);
+      }
```

Comment:
what's the point of initializating a scaleArray that no-one will know about ?

In my comment [1], I was not suggesting to create missing arrays, but instead "abort" or use a default "scale" value if the scale array is missing. (e.g. `const scale = scaleArray ? scaleArray.getTuple(i) : [10, 10, 1000]`)

[1]: https://github.com/Kitware/vtk-js/pull/3102/files#r1695095549

Reply:
Oh ok, this is fixed!